presto-main/src/main/java/com/facebook/presto/operator/PagesSpatialIndexSupplier.java

```diff
@@ -77,6 +81,13 @@ public PagesSpatialIndexSupplier(
         this.memorySizeInBytes = INSTANCE_SIZE +
```

Comment:
you can extract a method for this complex instance size calculation, which will simplify the constructor.